sklearn/preprocessing/tests/test_label.py

```diff
@@ -307,10 +308,12 @@ def test_multilabel_binarizer_empty_sample():
 def test_multilabel_binarizer_unknown_class():
     mlb = MultiLabelBinarizer()
     y = [[1, 2]]
-    assert_raises(KeyError, mlb.fit(y).transform, [[0]])
+    Y = np.array([[0, 0]])
+    assert_warns(UserWarning, mlb.fit(y).transform, [[0]])
+    assert_array_equal(mlb.fit(y).transform([[0]]), Y)
 
     mlb = MultiLabelBinarizer(classes=[1, 2])
-    assert_raises(KeyError, mlb.fit_transform, [[0]])
+    assert_array_equal(mlb.fit(y).transform([[0]]), Y)
```

Comment:
We don't want this raising the warning during testing. You can instead get the return value from assert_warns

Reply:
done